plugins/rest/rest_handlers.go

```diff
@@ -29,175 +29,111 @@ import (
-
-		plugin.Log.Debug("Getting list of all fibs")
-
-		// create an API channel
-		ch, err := plugin.GoVppmux.NewAPIChannel()
-		if err != nil {
-			plugin.Log.Errorf("Error creating channel: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		defer ch.Close()
-
-		fibHandler, err := l2plugin.NewFibVppHandler(ch, nil, make(chan *l2plugin.FibLogicalReq), plugin.Log, nil)
-		if err != nil {
-			plugin.Log.Errorf("Error creating VPP handler: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		res, err := fibHandler.DumpFIBTableEntries()
-		if err != nil {
-			plugin.Log.Errorf("Error: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, nil)
-			return
-		}
-
-		plugin.Log.Debug(res)
-		formatter.JSON(w, http.StatusOK, res)
-	}
+// Registers L2 plugin REST handlers
+func (plugin *Plugin) registerL2Handlers() {
+	// GET bridge domain IDs
+	plugin.registerHTTPHandler(resturl.BdId, GET, func() (interface{}, error) {
+		return plugin.bdHandler.DumpBridgeDomainIDs()
+	})
+	// GET bridge domains
+	plugin.registerHTTPHandler(resturl.Bd, GET, func() (interface{}, error) {
+		return plugin.bdHandler.DumpBridgeDomains()
+	})
+	// GET FIB entries
+	plugin.registerHTTPHandler(resturl.Fib, GET, func() (interface{}, error) {
+		return plugin.fibHandler.DumpFIBTableEntries()
+	})
+	// GET cross connects
+	plugin.registerHTTPHandler(resturl.Xc, GET, func() (interface{}, error) {
+		return plugin.xcHandler.DumpXConnectPairs()
+	})
 }
 
-// xconnectPairsGetHandler - used to get list of all connect pairs (transmit and receive interfaces)
-func (plugin *Plugin) xconnectPairsGetHandler(formatter *render.Render) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
+// registerHTTPHandler is common register method for all handlers
+func (plugin *Plugin) registerHTTPHandler(key, method string, f func() (interface{}, error)) {
+	handlerFunc := func(formatter *render.Render) http.HandlerFunc {
+		return func(w http.ResponseWriter, req *http.Request) {
+			plugin.Lock()
+			defer plugin.Unlock()
 
-		plugin.Log.Debug("Getting list of all xconnect pairs")
-
-		// create an API channel
-		ch, err := plugin.GoVppmux.NewAPIChannel()
-		if err != nil {
-			plugin.Log.Errorf("Error creating channel: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		defer ch.Close()
-
-		xcHandler, err := l2plugin.NewXConnectVppHandler(ch, plugin.Log, nil)
-		if err != nil {
-			plugin.Log.Errorf("Error creating VPP handler: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		res, err := xcHandler.DumpXConnectPairs()
-		if err != nil {
-			plugin.Log.Errorf("Error: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, nil)
+			res, err := f()
+			if err != nil {
+				plugin.Deps.Log.Errorf("Error: %v", err)
+				w.Write([]byte("500 Internal server error: " + err.Error()))
```

Comment:
This is going to write text into output and next line is going to print JSON. That is not very compatible.